protocols/interfaces.go

```diff
@@ -0,0 +1,51 @@
+package protocols
+
+import (
+	"math/big"
+
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// SideEffects is an list of effects to be executed by an imperative shell
+type SideEffects []interface{}
+
+// WaitingFor is an enumerable "pause-point" computed from an Objective. It describes how the objective is blocked on actions by third parties (i.e. co-participants or the blockchain).
+type WaitingFor string
```

Comment:
Seems feasible to me that objectives could be waiting for multiple things - eg, a direct funder could be waiting on counterpary signature and also counterparty's initial deposit, evidence of which might arrive in either order. Existing methods seem to be expecting a single `WaitingFor`.

Is this managed via finer-grained intermediary objectives?

Reply:
It's true that an objective is generally waiting on more than one thing. Perhaps `WaitingFor` could be an array of strings, with each one being struck off until the list is empty?